Block/Buttons/AbstractButton.php

```diff
@@ -255,4 +254,78 @@ public function getContainerId(): string
     {
         return $this->getData(self::BUTTON_ELEMENT_INDEX) ?: '';
     }
+
+    public function getRandomElementId(): string
+    {
+        try {
+            $id = sprintf('%s%s', $this->getContainerId(), random_int(PHP_INT_MIN, PHP_INT_MAX));
+        } catch (Exception $e) {
+            /**
+             * Exception only thrown if an appropriate source of randomness cannot be found.
+             * https://www.php.net/manual/en/function.random-int.php
+             */
+            $id = "0";
+        }
+
+        return $id;
+    }
+
+    /**
+     * Current Quote ID for guests
+     *
+     * @return string
+     * @throws LocalizedException
+     * @throws NoSuchEntityException
+     */
+    public function getQuoteId(): string
+    {
+        try {
+            $config = $this->defaultConfigProvider->getConfig();
+            if (!empty($config['quoteData']['entity_id'])) {
+                return $config['quoteData']['entity_id'];
+            }
+        } catch (NoSuchEntityException $e) {
+            if ($e->getMessage() !== 'No such entity with cartId = ') {
+                throw $e;
+            }
+        }
+        return '';
+    }
+
+    /**
+     * Returns Adyen payment method variant
+     *
+     * @return string
+     */
+    public function getPaymentMethodVariant(): string
+    {
+        return static::PAYMENT_METHOD_VARIANT;
+    }
+
+    /**
+     * Returns the base configuration for express frontend
+     *
+     * @return array[]
+     * @throws LocalizedException
+     * @throws NoSuchEntityException
+     */
+    public function buildConfiguration(): array
+    {
+        $variant = $this->getPaymentMethodVariant();
+
+        return [
+            "Adyen_ExpressCheckout/js/$variant/button" => [
+                'actionSuccess' => $this->getActionSuccess(),
+                'storeCode' => $this->getStorecode(),
+                'countryCode' => $this->getDefaultCountryCode(),
+                'currency' => $this->getCurrency(),
+                'merchantAccount' => $this->getMerchantAccount(),
+                'format' => $this->getFormat(),
+                'locale' => $this->getLocale(),
+                'originkey' => $this->getOriginKey(),
```

Comment:
It looks like `originkey` field has been removed from the frontend. Do we still need it here?

Reply:
We cannot remove it, GooglePay still uses this value for clientKey.